src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -24,11 +26,14 @@ public class SnippetsProvider : ISnippetsProvider
     {
         private const string RequiredPropertiesDescription = "Required properties";
         private const string RequiredPropertiesLabel = "required-properties";
+        private static readonly Regex ParentPropertyPattern = new Regex(@"^.*parent:.*$[\r\n]*", RegexOptions.Compiled | RegexOptions.Multiline);
 
-        // Used to cache resource declarations. Maps resource type to body text and description
-        private readonly ConcurrentDictionary<string, (string text, string description)> resourceTypeToBodyMap = new();
+        // Used to cache resource declaration information. Maps resource type to prefix, identifier, body text and description
+        private readonly ConcurrentDictionary<string, (string prefix, string identifier, string bodyText, string description)> resourceTypeInfoMap = new();
         // Used to cache resource dependencies. Maps resource type to it's dependencies
         private readonly ConcurrentDictionary<string, string> resourceTypeToDependentsMap = new();
+        // Used to cache information about child type symbols in nested resource scenario. Maps resource type to nested type symbols
+        private readonly ConcurrentDictionary<string, List<TypeSymbol>> resourceTypeToChildTypeSymbolsMap = new();
```

Comment:
I'd recommend instead using `ResourceTypeReference` as the dictionary key; you can then instantiate it with `new(ResourceTypeReferenceComparer.Instance)`

Reply:
Updated here:
https://github.com/Azure/bicep/commit/0c0c664878dbd906644c181a547c42f854f4d5f3